src/main/java/net/minecraftforge/common/world/ModifiableBiomeInfo.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.world;
+
+import java.util.List;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.core.Holder;
+import net.minecraft.world.level.biome.Biome;
+import net.minecraft.world.level.biome.Biome.ClimateSettings;
+import net.minecraft.world.level.biome.BiomeGenerationSettings;
+import net.minecraft.world.level.biome.BiomeSpecialEffects;
+import net.minecraft.world.level.biome.MobSpawnSettings;
+
+/**
+ * Holds lazy-evaluable modified biome info.
+ * Memoizers are not used because it's important to return null
+ * without evaluating the biome info if it's accessed outside of a server context.
+ */
+public class ModifiableBiomeInfo
+{
+    private final BiomeInfo originalBiomeInfo;
+    @Nullable
+    private BiomeInfo modifiedBiomeInfo = null;
+    
+    /**
+     * @param originalBiomeInfo BiomeInfo representing the original state of a biome when the biome was constructed.
+     */
+    public ModifiableBiomeInfo(final BiomeInfo originalBiomeInfo)
```

Comment:
Probably `@NotNull` would be good here and on the field and on getOriginalBiomeInfo.